### PR TITLE
Updates for DAGMC 3.2.0

### PR DESCRIPTION
**Pull request by pshriwise**
A couple of small updates to the DAGMC metadata interface require a couple of changes in our DAGMC code here. They're reflective of some improvements toward using modern C++ by @bam241, so overall I think they're a positive change. We're having discussions about the metadata handling in DAGMC soon. Part of that discussion will have to do with insulating against future interface changes like these.

I've included a check in the `CMakeLists.txt` for the DAGMC version here as well to avoid build errors for older versions of DAGMC without these updates. The DAGMC files for the `uwuw` and `refl` tests have updated because the material IDs in the UWUW file are now being read in a different order. This doesn't affect transport (hence no change to the expected results), but it does change the inputs slightly. I've adjusted the DAGMC files accordingly to account for that.

The other change to the material name comparison is a fix so that names are compared in lowercase. This is a regression that was reported by @helen-brooks a while ago that I'm just getting around to.

### CODE REVIEW

**Comment by Shimwell** [comment]
Ah super, perfect timing, I just came across a build error that I think this PR will fix

```
[  7%] Building CXX object CMakeFiles/libopenmc.dir/src/bank.cpp.o
[  8%] Building CXX object CMakeFiles/libopenmc.dir/src/boundary_condition.cpp.o
[  9%] Building CXX object CMakeFiles/libopenmc.dir/src/bremsstrahlung.cpp.o
[ 10%] Building CXX object CMakeFiles/libopenmc.dir/src/dagmc.cpp.o
[ 11%] Building CXX object CMakeFiles/libopenmc.dir/src/cell.cpp.o
[91m/opt/openmc/src/dagmc.cpp: In function â€˜bool openmc::get_uwuw_materials_xml(std::__cxx11::string&)â€™:
/opt/openmc/src/dagmc.cpp:69:49: error: â€˜class std::shared_ptr<pyne::Material>â€™ has no member named â€˜openmcâ€™
     for (auto mat : mat_lib) { ss << mat.second.openmc("atom"); }
                                                 ^~~~~~
[0m[91m/opt/openmc/src/dagmc.cpp: In function â€˜void openmc::load_dagmc_geometry()â€™:
/opt/openmc/src/dagmc.cpp:242:60: error: â€˜class std::shared_ptr<pyne::Material>â€™ has no member named â€˜metadataâ€™
           int mat_number = uwuw.material_library[uwuw_mat].metadata["mat_number"].asInt();
                                                            ^~~~~~~~
[0m[91mmake[2]: *** [CMakeFiles/libopenmc.dir/build.make:121: CMakeFiles/libopenmc.dir/src/dagmc.cpp.o] Error 1
[0m[91mmake[2]: *** Waiting for unfinished jobs....
[0m[91mmake[1]: *** [CMakeFiles/Makefile2:267: CMakeFiles/libopenmc.dir/all] Error 2
[0m[91mmake: *** [Makefile:149: all] Error 2
The command '/bin/sh -c cd /opt &&     git clone --single-branch --branch develop https://github.com/openmc-dev/openmc.git &&     cd openmc &&     mkdir build &&     cd build &&     cmake -Doptimize=on -Ddagmc=ON -DDAGMC_ROOT=$DAGMC_INSTALL_DIR -DHDF5_PREFER_PARALLEL=OFF ..  &&     make -j"$compile_cores" &&     make -j"$compile_cores" install &&     cd /opt/openmc/ &&     pip install .' returned a non-zero code: 2
ERROR
ERROR: build step 0 "gcr.io/cloud-builders/docker" failed: step exited with non-zero status: 2
[0m
```

I am just retrying the build using your branch and will report back

**Comment by pshriwise** [comment]
Yeah, it should hopefully take care of those. Thanks for trying it out!

**Comment by Shimwell** [comment]
I can confirm it builds over here

File: CMakeLists.txt [hunk 1]
```diff
@@ -40,6 +40,10 @@ endif()
 #===============================================================================
 if(dagmc)
   find_package(DAGMC REQUIRED PATH_SUFFIXES lib/cmake)
+  if (${DAGMC_VERSION} VERSION_LESS 3.2.0)
+    message(FATAL_ERROR "Discovered DAGMC Version: ${DAGMC_VERSION}. \
+    Please update DAGMC to version 3.2.0 or greater.")
```
[the remark this answers is not in the document]
Comment (pshriwise) [reply CMakeLists.txt:45]: If there were a `DAGMCConfig-version.cmake` file, yes. But one hasn't been added as of yet.

[the remark this answers is not in the document]
Comment (paulromano) [reply CMakeLists.txt:45]: Gotcha; thanks!

**Review by paulromano — APPROVED** [review]

**Comment by paulromano** [comment]
Thanks for the fix @pshriwise!

**Pull request merged**